hls-plugin-api/src/Ide/Plugin/ConfigUtils.hs

```diff
@@ -0,0 +1,88 @@
+{-# LANGUAGE GADTs             #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+{-# LANGUAGE ViewPatterns      #-}
+
+module Ide.Plugin.ConfigUtils where
+
+import qualified Data.Aeson            as A
+import qualified Data.Aeson.Types      as A
+import           Data.Default          (def)
+import qualified Data.Dependent.Map    as DMap
+import qualified Data.Dependent.Sum    as DSum
+import qualified Data.HashMap.Lazy     as HMap
+import qualified Data.Map              as Map
+import           Ide.Plugin.Config
+import           Ide.Plugin.Properties (toDefaultJSON, toVSCodeExtensionSchema)
+import           Ide.Types
+import           Language.LSP.Types
+
+pluginsToDefaultConfig :: IdePlugins a -> A.Value
+pluginsToDefaultConfig IdePlugins {..} =
+  A.Object $
+    HMap.adjust
+      ( \(unsafeValueToObject -> o) ->
+          A.Object $ HMap.insert "plugin" elems o
+      )
+      "haskell"
+      (unsafeValueToObject (A.toJSON defaultConfig))
+  where
+    defaultConfig@Config {} = def
+    unsafeValueToObject (A.Object o) = o
+    unsafeValueToObject _            = error "impossible"
```

Comment:
This makes me unhappy --- is there a safer way to write this?

Reply:
The idea is to reuse the default values of `Config` defined here: https://github.com/haskell/haskell-language-server/blob/a34d927ae152ae117eade79f6c4781faac338ff7/hls-plugin-api/src/Ide/Plugin/Config.hs#L64

However, we need inplace our generated `plugins` which has type `Value` instead of `Map T.Text PluginConfig`, so I think it's inevitable to deconstruct `Value` and operate the `HashMap.`

Reply:
My concern is more about the `error` call. Is it really impossible? It's not immediately clear to me why that would be so

Reply:
`unsafeValueToObject` is applied to `defaultConfig`:
```json
{
    "haskell": {
        "hlintOn": true,
        "formatOnImportOn": true,
        "checkParents": "CheckOnSaveAndClose",
        "liquidOn": false,
        "maxCompletions": 40,
        "checkProject": true,
        "diagnosticsOnChange": true,
        "completionSnippetsOn": true,
        "diagnosticsDebounceDuration": 350000,
        "formattingProvider": "ormolu",
        "plugin": {}
    }
}
```
and the `"haskell"` section:
```json
{
    "hlintOn": true,
    "formatOnImportOn": true,
    "checkParents": "CheckOnSaveAndClose",
    "liquidOn": false,
    "maxCompletions": 40,
    "checkProject": true,
    "diagnosticsOnChange": true,
    "completionSnippetsOn": true,
    "diagnosticsDebounceDuration": 350000,
    "formattingProvider": "ormolu",
    "plugin": {}
}
```

both of which are definitely `Object`s.